shared/src/main/scala/amf/client/remod/AMFConfiguration.scala

```diff
@@ -0,0 +1,104 @@
+package amf.client.remod
+
+import amf.client.remod.amfcore.config._
+import amf.client.remod.amfcore.plugins.AMFPlugin
+import amf.client.remod.amfcore.plugins.parse.AMFParsePlugin
+import amf.client.remod.amfcore.registry.AMFRegistry
+import amf.core.validation.core.ValidationProfile
+import amf.internal.environment.Environment
+import amf.internal.reference.UnitCache
+import amf.internal.resource.ResourceLoader
+
+import scala.concurrent.ExecutionContext
+// all constructors only visible from amf. Users should always use builders or defaults
+
+class AMFConfiguration(private[amf] val resolvers: AMFResolvers,
+                       private[amf] val errorHandlerProvider: ErrorHandlerProvider,
+                       private[amf] val registry: AMFRegistry,
+                       private[amf] val logger: AMFLogger,
+                       private[amf] val listeners: List[AMFEventListener],
+                       private[amf] val options: AMFOptions) { // break platform into more specific classes?
+
+  def createClient(): AMFClient = new AMFClient(this)
+
+  def withParsingOptions(parsingOptions: ParsingOptions): AMFConfiguration =
+    copy(options = options.copy(parsingOptions = parsingOptions))
+
+  def withRenderOptions(renderOptions: RenderOptions): AMFConfiguration =
+    copy(options = options.copy(renderOptions = renderOptions))
+
+  def withErrorHandlerProvider(provider: ErrorHandlerProvider): AMFConfiguration =
+    copy(errorHandlerProvider = provider)
+
+  def withResourceLoader(rl: ResourceLoader): AMFConfiguration = copy(resolvers.withResourceLoader(rl))
+
+  def withResourceLoaders(rl: List[ResourceLoader]): AMFConfiguration =
+    copy(resolvers = resolvers.withResourceLoaders(rl))
+
+  def withUnitCache(cache: UnitCache): AMFConfiguration =
+    copy(resolvers.withCache(cache))
+
+  def withPlugin(amfPlugin: AMFParsePlugin): AMFConfiguration = copy(registry = registry.withPlugin(amfPlugin))
+
+  def withPlugins(plugins: List[AMFPlugin[_]]): AMFConfiguration = copy(registry = registry.withPlugins(plugins))
+
+  // TODO: delete after branch?
+  def removePlugin(id: String): AMFConfiguration = copy(registry = registry.removePlugin(id))
+
+  def withConstraints(profile: ValidationProfile): AMFConfiguration =
```

Comment:
Troublesome that the method says `constraints` but the param is a `ValidationProfile` while the rest of the methods match this relationship.